src/main/java/io/coala/jetbrains/analysis/CodeAnalysisRunner.java

```diff
@@ -0,0 +1,147 @@
+     *
+     * @return the instance with all the required information about the run
+     * @throws ExecutionException
+     */
+    public ProcessOutput analyze() throws ExecutionException {
+        final GeneralCommandLine commandLine = getNewGeneralCommandLine();
+
+        final String commandLineString = commandLine.getCommandLineString();
+        final Process process = commandLine.createProcess();
+
+        LOGGER.info("Running coala command " + commandLineString);
+
+        final OSProcessHandler processHandler = new OSProcessHandler(process, commandLineString);
+
+        ProcessOutput processOutput = getProcessOutputWithTextAvailableListener(processHandler);
+
+        processHandler.startNotify();
+
+        holdAndWaitProcess(processHandler, processOutput);
+
+        final String stdout = processOutput.getStdout();
+        LOGGER.warn(stdout);
+
+        LOGGER.info("Finished running coala.");
+
+        return processOutput;
+    }
+
+    /**
+     * This method creates a new instance of GeneralCommandLine
+     * and sets its configurations according to the input params provided
+     * <p>
+     * The following configurations are set:
+     * the current working directory
+     * the executable path
+     * and the following parameters
+     * --filter-by specifies which sections (in .coafile) to run coala on
+     * --json specifies the output should be in JSON format
+     * --log-json specifies the logging statements must be in JSON as well
+     *
+     * @return an instance of GeneralCommandLine
+     */
+    public GeneralCommandLine getNewGeneralCommandLine() {
+        final String cwd = projectSettings.getCwd();
+        final String executable = projectSettings.getExecutable();
+        final List<String> sections = projectSettings.getSections();
+
+        final GeneralCommandLine commandLine = new GeneralCommandLine();
+        commandLine.setWorkDirectory(cwd);
+        commandLine.setExePath(executable);
+        commandLine.addParameter("--json");
+        commandLine.addParameter("--log-json");
+
+        commandLine.addParameters("--filter-by", "section_tags");
+        for(String section : sections) {
+            commandLine.addParameters(section);
+        }
+
+        return commandLine;
+    }
+
+    /**
+     * This method waits for the process to exit and subsequently
+     * sets the required flags for {@link ProcessOutput}
+     *
+     * @param processHandler the instance of running process
+     * @param processOutput the instance to set appropriate flags
+     */
+    private void holdAndWaitProcess(@NotNull OSProcessHandler processHandler, @NotNull ProcessOutput processOutput) {
+        if(processHandler.waitFor(timeOutInMilliseconds)) {
+            LOGGER.info("Process exited with exit code " + processHandler.getExitCode());
+            processOutput.setExitCode(processHandler.getExitCode());
+        } else {
+            LOGGER.info("Thanos finger snap!");
```

Comment:
He missed this comment though.

Reply:
Since he left it, maybe I leave this as it is too? Maybe as an easter egg? XD

Reply:
Sure. I know the consequences when you cross the big man.